CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Minimum cmake version required by scikit-build-core: 3.15

```

Comment:
I feel we can remove this and only keep the 3.17 comment

Reply:
I just realized we can set it to 3.15. If the version is lower than 3.26, `scikit-build-core` will ensure FindPython is backported: https://github.com/scikit-build/scikit-build-core